src/util/docker.rs

```diff
@@ -51,6 +56,31 @@ pub struct ContainerImage {
     pub short_name: ImageName,
 }
 
+// To convert a user-supplied image name into an expanded image name:
+pub fn resolve_image_name(name: &str, available_images: &Vec<ContainerImage>) -> Result<ImageName> {
+    let mut images = HashMap::new();
+    for image in available_images {
+        if images.insert(&image.name, &image.name).is_some() {
+            warn!(
+                "The image name \"{0}\" is specified multiple times in the configured `images` list",
+                image.name
+            );
+        }
+        if images.insert(&image.short_name, &image.name).is_some() {
+            warn!(
+                "The image short name \"{0}\" is specified multiple times in the configured `images` list",
+                image.short_name
+            );
+        }
+    }
+    images.get(&ImageName::from(name.to_string())).cloned().ok_or_else(|| {
+        let mut available_images = images.into_keys().map(|name| name.0.to_string()).collect::<Vec<_>>();
+        available_images.sort_unstable();
+        let available_images = available_images.join(",");
+        anyhow!("Failed to resolve the requested container image name \"{name}\". The available images are: {available_images}")
```

Comment:
Depending on personal preferences we could consider turning this into a warning or turning the two warnings above into hard errors.